core/src/main/scala/com.snowplowanalytics.stream/loader/StreamLoaderApp.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2014-2017 Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache
+ * License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ *
+ * See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.stream.loader
+
+// Java
+import java.io.File
+
+// Config
+import com.typesafe.config.ConfigFactory
+
+// Scala
+import scopt.OptionParser
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Pureconfig
+import pureconfig._
+
+// This project
+import sinks._
+import model._
+
+/**
+ * Main entry point for the Elasticsearch sink
+ */
+trait StreamLoaderApp extends App {
```

Comment:
could we move away from using `App`?

Reply:
This is from the current code. If I have done it in this PR, you would have added to it the too much work list :stuck_out_tongue_winking_eye: Also, as you know `App` has internal mechanisms that removing it, might not look straight-forward and with small change.